kw_sdk.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+// TODO: figure out if it's worth to move this to a dedicated library
+// We cannot add that to the Kubewarden Go SDK because it would not work
+// with TinyGo
+
+import (
+	"encoding/json"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// ValidationRequest describes the input received by the policy
+// when invoked via the `validate` subcommand
+type ValidationRequest struct {
+	Request     admissionv1.AdmissionRequest `json:"request"`
+	SettingsRaw json.RawMessage              `json:"settings"`
+}
+
+// Message is the optional string used to build validation responses
+type Message string
```

Comment:
I see the benefits of using a contextual name for a type. Thus, why not using this type inside the `ValidationResponse` and `SettingsValidationResponse` types in the `Message` field?

Reply:
BTW, this is copy and pasted with some minor changes from the official go sdk.

I don't know why we didn't do that. I think the reasoning was that regular users create these objects with the helper function we provide (like `RejectRequest`); these functions hint good practices by using this type.
When dealing with "low level" objects, we just use vanilla types for simplicity.

Is it worth to change that? I fear this would cause some renaming, otherwise we would end up with something like that:

```golang
type ValidationResponse {
  // other fields
  Message *Message
}
```

which is a bit weird IMHO

Reply:
Okay. Fine by me. Let's keep it as it is now.